Sources/NIO/ChannelPipeline.swift

```diff
@@ -931,10 +931,11 @@ extension ChannelPipeline {
 }
 
 private extension CloseMode {
+    /// Returns the error to fail outstanding operations writes with.
     var error: ChannelError {
         switch self {
         case .all:
-            return .alreadyClosed
+            return .ioOnClosedChannel
```

Comment:
this was just the wrong error here, `.alreadyClosed` means close on previously closed channel. For all others, `.ioOnClosedChannel` is right